File: cpp/bubble.cpp
#include <stdio.h>

#include <iostream>
#include <vector>

using namespace std;

void print_all(const vector<int>& arr)
{
    for (size_t i = 0; i < arr.size(); i++)
        cout << arr[i] << endl;
}

int main()
{
    const auto n = 40000;
    auto arr = vector<int>(40000);

    for (size_t i = 0; i < arr.size(); i++)
        arr[i] = n - i; //random()%100500;

    print_all(arr);
    printf("---\n");

    for (size_t i = 0; i < arr.size(); i++) {
        for (size_t j = 0; j < arr.size() - 1 - i; j++) {
            auto tmp = arr[j];
            if (arr[j] > arr[j + 1]) {
                arr[j] = arr[j + 1];
                arr[j + 1] = tmp;
            }
        }
    }

    print_all(arr);
    return 0;
}

File: rust/bubble.rs
fn print_all(arr: &[i32]) {
    for i in 0..arr.len() {
        println!("{}", arr[i]); 
    }    
}


fn main() {
    const N: usize = 40000;
    let mut arr: [i32; N] = [0; N];

    for i in 0..arr.len() { arr[i] = (N-i) as i32; }

    print_all(&arr);
    println!("-----"); 

    for i in 0..arr.len() {
        for j in 0..arr.len() - 1 - i {
            let tmp = arr[j];
            if  arr[j] > arr[j + 1] {
                arr[j] = arr[j + 1];
                arr[j + 1] = tmp;
            }
        }
    }
    print_all(&arr);
}
